corebuild/integration/test/IntegrationTestBase.cs

```diff
@@ -101,38 +111,37 @@ protected int RunCommand(string command, string args, string workingDir, string
 		///   linker task package.
 		///   Optionally takes a list of root descriptor files.
 		/// </summary>
-		public void BuildAndLink(string csproj, List<string> rootFiles = null)
+		public void BuildAndLink(string csproj, List<string> rootFiles = null, Dictionary<string, string> extraPublishArgs = null)
 		{
 			string rid = context.RuntimeIdentifier;
 			string config = context.Configuration;
 			string demoRoot = Path.GetDirectoryName(csproj);
 
-			int ret = Dotnet($"restore -r {rid}", demoRoot);
-			if (ret != 0) {
-				output.WriteLine("restore failed");
-				Assert.True(false);
-				return;
-			}
-
```

Comment:
Why is restore no longer needed?

Reply:
With the new SDK, publish implicitly does a restore.